paddle/fluid/ir_adaptor/translator/op_translator.cc

```diff
@@ -1863,6 +1863,110 @@ struct FillConstantTranscriber : public OpTranscriber {
   }
 };
 
+static std::vector<int64_t> ParseCompatibleShapes(
+    const std::vector<int64_t>& dim1, const std::vector<int64_t>& dim2) {
+  IR_ENFORCE(dim1.size() == dim2.size(),
+             "Does not support rank inconsistency: dim1=%d, dim2=%d",
+             dim1.size(),
+             dim2.size());
+  std::vector<int64_t> result;
+  for (size_t i = 0; i < dim1.size(); ++i) {
+    if (dim1[i] != dim2[i]) {
+      result.push_back(-1);
+    } else {
+      result.push_back(dim1[i]);
+    }
+  }
+  return result;
+}
+
+struct SelectInputOpTranscriber : public OpTranscriber {
+  pir::Operation* operator()(pir::IrContext* ctx,
+                             TranslationContext* param_map,
+                             const OpDesc& op_desc,
+                             pir::Block* block) override {
+    VLOG(10) << "[op select_input] start transcribing";
+    auto op_info = this->LoopkUpOpInfo(ctx, op_desc);
+
```

Comment:
建议在这里添加一句 
```suggestion
this->InsertSliceOperationForInput(ctx, param_map, op_desc, input_infos, block);
```
保证输入的类型正确